application/github/v0/pull_request.go

```diff
@@ -0,0 +1,174 @@
+package github
+
+import (
+	"context"
+
+	"fmt"
+
+	"github.com/google/go-github/v62/github"
+	"github.com/instill-ai/component/base"
+	"github.com/instill-ai/x/errmsg"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type PullRequestService interface {
+	List(context.Context, string, string, *github.PullRequestListOptions) ([]*github.PullRequest, *github.Response, error)
+	Get(context.Context, string, string, int) (*github.PullRequest, *github.Response, error)
+	ListComments(context.Context, string, string, int, *github.PullRequestListCommentsOptions) ([]*github.PullRequestComment, *github.Response, error)
+	CreateComment(context.Context, string, string, int, *github.PullRequestComment) (*github.PullRequestComment, *github.Response, error)
+	ListCommits(context.Context, string, string, int, *github.ListOptions) ([]*github.RepositoryCommit, *github.Response, error)
+}
+
+type PullRequest struct {
+	ID          		int64           	`json:"id"`
+	Number      		int             	`json:"number"`
+	State       		string          	`json:"state"`
+	Title	 			string 				`json:"title"`
+	Body 				string 				`json:"body"`
+	DiffURL 			string 				`json:"diff_url,omitempty"`
+	CommitsURL 			string 				`json:"commits_url,omitempty"`
+	Commits 			[]Commit 			`json:"commits"`
+	Head 				string 				`json:"head"`
+	Base 				string 				`json:"base"`
+	CommentsNum 		int 				`json:"comments_num"`
+	CommitsNum 			int 				`json:"commits_num"`
+	ReviewCommentsNum 	int		 			`json:"review_comments_num"`
+}
+func (githubClient *Client) extractPullRequestInformation(originalPr *github.PullRequest) (PullRequest, error) {
+	resp := PullRequest{
+		ID: originalPr.GetID(),
+		Number: originalPr.GetNumber(),
+		State: originalPr.GetState(),
+		Title: originalPr.GetTitle(),
+		Body: originalPr.GetBody(),
+		DiffURL: originalPr.GetDiffURL(),
+		Head: originalPr.GetHead().GetSHA(),
+		Base: originalPr.GetBase().GetSHA(),
+		CommentsNum: originalPr.GetComments(),
+		CommitsNum: originalPr.GetCommits(),
+		ReviewCommentsNum: originalPr.GetReviewComments(),
+	}
+	if originalPr.GetCommitsURL() != "" {
+		commits, _, err := githubClient.client.PullRequests.ListCommits(context.Background(), githubClient.owner, githubClient.repository, resp.Number, nil)
+		if err != nil {
+			return PullRequest{}, err
+		}
+		resp.Commits = make([]Commit, len(commits))
+		for idx, commit := range commits {
+			resp.Commits[idx] = githubClient.extractCommitInformation(commit)
+		}
+	}
+	return resp, nil
+}
+
+// for testing
```

Comment:
NIT
If it is only for test, we probably want it to be in the test file.

Reply:
Due to [this change](https://github.com/instill-ai/component/pull/177#discussion_r1664228934), it's not only for test now.